apps/web/src/pages/quick-start/components/useInAppActivated.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useState } from 'react';
+import { getInAppActivated } from '../../../api/integration';
+
+export function useInAppActivated() {
+  const [inAppInitialized, setInAppInitialized] = useState<boolean>(false);
+  let intervalId: NodeJS.Timer;
+  const REQUEST_INTERVAL = 3000;
+
+  useEffect(() => {
+    setInit();
+
+    intervalId = setInterval(() => setInit(), REQUEST_INTERVAL);
+  }, []);
+
+  async function setInit() {
+    const activeRes = await getInAppActivated();
+
+    if (activeRes) {
+      clearInterval(intervalId);
+      setInAppInitialized(activeRes);
+    }
+  }
+
+  return { initialized: inAppInitialized };
```

Comment:
I think you could actually use react query with refetch interval instead if implementing this yourself. Wdyt?

Reply:
I am still too much into the DIY 😅 I did not know that it has refetch by interval, thanks for the heads up!